src/main/java/com/google/flourbot/api/DriveCloudSheet.java

```diff
@@ -10,51 +10,104 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Collections;
 
 public class DriveCloudSheet implements CloudSheet {
-  Sheets sheetsService;
-  String spreadsheetId;
+  private static final String VALUE_INPUT_OPTION = "USER_ENTERED";
+  private static final String INSERT_DATA_OPTION = "INSERT_ROWS";
+  private Sheets sheetsService;
+  private String spreadsheetId;
 
   public DriveCloudSheet(String documentId, Sheets documentService) {
     this.spreadsheetId = documentId;
     this.sheetsService = documentService;
   }
 
-  // TODO: replace hardcoded range with calculated range (so far values must be length 3)
-  public void appendRow(List<String> values) throws IOException, GeneralSecurityException {
+  private String getRange(int numEntryColumns, int numSkipColumns) {
+    // Returns range for sheets request by calculating the alphabetic name of
+    // the start and end column based on the number of columns filled and pre-skipped
+
+    String startColumn = toAlphabetic(numSkipColumns);
+    String endColumn = toAlphabetic(numSkipColumns + numEntryColumns - 1);
+    return String.format("Sheet1!%s:%s", startColumn, endColumn);
+  }
+
+  private String toAlphabetic(int i) {
+    // Recursive strategy to build a string that represents the alphabetic name
+    // of a column based on column number i
+    // e.g. 0 -> A, 1 -> B... 25 -> Z, 26 -> AA...
     
-    // Generate request to append to sheet
-    String range = "Sheet1!A1:C1";
-    String valueInputOption = "USER_ENTERED";
-    String insertDataOption = "INSERT_ROWS";
+    // Negative numbers cause errors
+    if (i < 0) {
+      throw new IllegalArgumentException(String.format("Column number %d is invalid", i));
+    }
+
+    // Every calculation produces 1 letter
+    int quotient = i / 26;
+    char letter = (char)(i % 26 + 'A');
+    if (quotient == 0) {
+        return "" + letter;
+    } else {
+        return toAlphabetic(quotient-1) + letter;
+    }
+  }
+
+  public List<List<String>> readRange(String range) {
```

Comment:
What is this range and who defines it? If the format is A1:B3, it would be good to expose a method that builds this range for the caller if needed. Or have the method that takes startRow, endRow, startColumn, endColumn and build the range, similar to what you do with getColumn. Having a method that validates the range would be good as well. But all of these are good to have, no need to implement now.

Reply:
Ah I like the idea of having a method for it. One option is just to hope the user uses A1 notation but I think that causes more friction on their end so we should handle it